internal/hud/server/logs_reader.go

```diff
@@ -67,28 +82,36 @@ func (ls *LogStreamer) Handle(v proto_webview.View) error {
 	}
 
 	segments := v.LogList.Segments
-	if fromCheckpoint < ls.checkpoint {
+	if fromCheckpoint < ls.checkpointExternal {
 		// The server is re-sending some logs we already have, so slice them off.
-		deleteCount := ls.checkpoint - fromCheckpoint
+		deleteCount := ls.checkpointExternal - fromCheckpoint
 		segments = segments[deleteCount:]
 	}
 
-	// TODO(maia): filter for the resources that we care about (`tilt logs resourceA resourceC`)
-	//   --> and if there's only one resource, don't prefix logs with resource name?
+	// TODO(maia): if there's only one resource, don't prefix logs with resource name
+	var segsAdded int
 	for _, seg := range segments {
-		// TODO(maia): secrets???
-		ls.logstore.Append(webview.LogSegmentToEvent(seg, v.LogList.Spans), model.SecretSet{})
+		var mn string
+		if span, ok := v.LogList.Spans[seg.SpanId]; ok {
+			mn = span.ManifestName
+		}
+		if noFilter || ls.resources[mn] {
+			// TODO(maia): secrets???
+			ls.logstore.Append(webview.LogSegmentToEvent(seg, v.LogList.Spans), model.SecretSet{})
+			segsAdded += 1
+		}
 	}
 
-	ls.printer.Print(ls.logstore.ContinuingLines(ls.checkpoint))
+	ls.printer.Print(ls.logstore.ContinuingLines(ls.checkpointInternal))
 
-	if toCheckpoint > ls.checkpoint {
-		ls.checkpoint = toCheckpoint
+	if segsAdded > 0 {
+		ls.checkpointInternal += logstore.Checkpoint(segsAdded)
```

Comment:
shouldn't this be `ls.logstore.Checkpoint()`, like we do in TerminalStream?

no reason to be doing checkpoint arithmetic out here